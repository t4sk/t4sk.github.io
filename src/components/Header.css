.Header-brand {
  display: flex;
  flex-direction: row;
  align-items: baseline;
}

.Header-brand-img {
  height: 50px;
  width: 50px;
  margin: 10px;
}
