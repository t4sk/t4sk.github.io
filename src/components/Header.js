import React from "react"
import PropTypes from "prop-types"
import Navbar from "react-bootstrap/Navbar"
import Nav from "react-bootstrap/Nav"
import Image from "react-bootstrap/Image"
import "./Header.css"
import Logo from "./Logo.svg"
import LanguageSVG from "./LanguageSVG"

export function Header(props) {
  const { lang } = props

  return (
    <Navbar expand="md" fixed>
      <Navbar.Brand href={`/${lang}`} className="Header-brand">
        <div className="Header-brand-img">
          <Image src={Logo} fluid />
        </div>
        SMART CONTRACT PROGRAMMER
      </Navbar.Brand>
      <Navbar.Toggle aria-controls="basic-navbar-nav" />
      <Navbar.Collapse id="basic-navbar-nav" className="justify-content-end">
        <Nav>
          <Nav.Link href={`/${lang}/blog`}>Blog</Nav.Link>
        </Nav>
        <Nav>
          <Nav.Link href={`/${lang}/about`}>About</Nav.Link>
        </Nav>
        <Nav>
          <Nav.Link href="/languages">
            <LanguageSVG />
            <span>Languages</span>
          </Nav.Link>
        </Nav>
      </Navbar.Collapse>
    </Navbar>
  )
}

Header.propTypes = {
  lang: PropTypes.oneOf(["en", "jp"]).isRequired,
}

export default Header
