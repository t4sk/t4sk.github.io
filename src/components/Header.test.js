import React from "react"
import { shallow } from "enzyme"
import Header from "./Header"

const props = {
  lang: "en",
}

test("en", () => {
  const component = shallow(<Header {...props} />)

  expect(component).toMatchSnapshot()
})

test("jp", () => {
  const component = shallow(<Header {...props} />)

  expect(component).toMatchSnapshot()
})
