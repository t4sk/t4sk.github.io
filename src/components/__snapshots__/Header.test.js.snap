// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`en 1`] = `
<Bootstrap(undefined)
  expand="md"
  fixed={true}
>
  <NavbarBrand
    className="Header-brand"
    href="/en"
  >
    <div
      className="Header-brand-img"
    >
      <Bootstrap(Image)
        fluid={true}
        src="Logo.svg"
      />
    </div>
    SMART CONTRACT PROGRAMMER
  </NavbarBrand>
  <NavbarToggle
    aria-controls="basic-navbar-nav"
    as="button"
    label="Toggle navigation"
  />
  <NavbarCollapse
    className="justify-content-end"
    id="basic-navbar-nav"
  >
    <Nav
      as="div"
      fill={false}
      justify={false}
    >
      <NavLink
        as={[Function]}
        disabled={false}
        href="/en/blog"
      >
        Blog
      </NavLink>
    </Nav>
    <Nav
      as="div"
      fill={false}
      justify={false}
    >
      <NavLink
        as={[Function]}
        disabled={false}
        href="/en/about"
      >
        About
      </NavLink>
    </Nav>
    <Nav
      as="div"
      fill={false}
      justify={false}
    >
      <NavLink
        as={[Function]}
        disabled={false}
        href="/languages"
      >
        <LangaugeSVG />
        <span>
          Languages
        </span>
      </NavLink>
    </Nav>
  </NavbarCollapse>
</Bootstrap(undefined)>
`;

exports[`it renders 1`] = `
<Bootstrap(undefined)
  expand="md"
  fixed={true}
>
  <NavbarBrand
    className="Header-brand"
    href="/"
  >
    <div
      className="Header-brand-img"
    >
      <Bootstrap(Image)
        fluid={true}
        src="Logo.svg"
      />
    </div>
    SMART CONTRACT PROGRAMMER
  </NavbarBrand>
  <NavbarToggle
    aria-controls="basic-navbar-nav"
    as="button"
    label="Toggle navigation"
  />
  <NavbarCollapse
    className="justify-content-end"
    id="basic-navbar-nav"
  >
    <Nav
      as="div"
      fill={false}
      justify={false}
    >
      <NavLink
        as={[Function]}
        disabled={false}
        href="/blog"
      >
        Blog
      </NavLink>
    </Nav>
    <Nav
      as="div"
      fill={false}
      justify={false}
    >
      <NavLink
        as={[Function]}
        disabled={false}
        href="/about"
      >
        About
      </NavLink>
    </Nav>
    <Nav
      as="div"
      fill={false}
      justify={false}
    >
      <NavLink
        as={[Function]}
        disabled={false}
        href="/languages"
      >
        <LangaugeSVG />
        <span>
          Languages
        </span>
      </NavLink>
    </Nav>
  </NavbarCollapse>
</Bootstrap(undefined)>
`;

exports[`jp 1`] = `
<Bootstrap(undefined)
  expand="md"
  fixed={true}
>
  <NavbarBrand
    className="Header-brand"
    href="/en"
  >
    <div
      className="Header-brand-img"
    >
      <Bootstrap(Image)
        fluid={true}
        src="Logo.svg"
      />
    </div>
    SMART CONTRACT PROGRAMMER
  </NavbarBrand>
  <NavbarToggle
    aria-controls="basic-navbar-nav"
    as="button"
    label="Toggle navigation"
  />
  <NavbarCollapse
    className="justify-content-end"
    id="basic-navbar-nav"
  >
    <Nav
      as="div"
      fill={false}
      justify={false}
    >
      <NavLink
        as={[Function]}
        disabled={false}
        href="/en/blog"
      >
        Blog
      </NavLink>
    </Nav>
    <Nav
      as="div"
      fill={false}
      justify={false}
    >
      <NavLink
        as={[Function]}
        disabled={false}
        href="/en/about"
      >
        About
      </NavLink>
    </Nav>
    <Nav
      as="div"
      fill={false}
      justify={false}
    >
      <NavLink
        as={[Function]}
        disabled={false}
        href="/languages"
      >
        <LangaugeSVG />
        <span>
          Languages
        </span>
      </NavLink>
    </Nav>
  </NavbarCollapse>
</Bootstrap(undefined)>
`;
